### PR TITLE
Cooldown Inhibitor Change Seconds To Formatted String

### PR DESCRIPTION
**Pull request by Skillz4Killz**
### Description of the PR
This PR changes the cooldown inhibitor to respond with a more user friendly string showing how much time is left. Currently, it always shows the number of seconds. However, for a command like a `!daily` command this looks really bad when it says like 86,395 seconds remaining. Using the Duration class this is converted into a more readable version.

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Fix cooldown timer.

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [ ] This PR fixes a bug and does not change the (intended) framework interface.
- [ ] This PR adds methods or properties to the framework interface.
- [ ] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

File: src/inhibitors/cooldown.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = class extends Inhibitor {
 
 		const existing = command.cooldowns.get(message.levelID);
 
-		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', Math.ceil(existing.remainingTime / 1000), command.cooldownLevel !== 'author');
+		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', Duration.toNow(Date.now() + Math.ceil(existing.remainingTime)), command.cooldownLevel !== 'author');
```
Comment (bdistin) [inline src/inhibitors/cooldown.js]: You don't need Math#ceil() if you aren't dividing.

**Comment by Skillz4Killz** [comment]
**Changes:**

1. Requested changes from Ao
2. I’ve altered the Duration.toNow method slightly to return a more accurate string in `DD HH MM SS` format example `1d 5h 3m 25s`. Let me know if this is the best way to handle it or it should be done differently please.

File: src/lib/util/Duration.js [hunk 2]
```diff
@@ -115,12 +115,21 @@ class Duration {
...
+	 * @param {boolean} [exact] Whether the output should be exact in DD-HH-MM-SS format
 	 * @returns {string}
 	 */
-	static toNow(earlier, showIn) {
+	static toNow(earlier, showIn, exact) {
 		if (!(earlier instanceof Date)) earlier = new Date(earlier);
 		const returnString = showIn ? 'in ' : '';
 		let duration = Math.abs((Date.now() - earlier) / 1000);
+		
+		if (exact) {
```
Comment (MrJacz) [inline src/lib/util/Duration.js]: this should be using timestamp

**Comment by Skillz4Killz** [comment]
**Changes:**

- Removed all changes on Duration.
- Created a separate method inside the cooldown inhibitor to convert seconds to string format DD-HH-MM-SS

File: src/inhibitors/cooldown.js [hunk 3]
```diff
@@ -11,7 +11,15 @@ module.exports = class extends Inhibitor {
...
-		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', Math.ceil(existing.remainingTime / 1000), command.cooldownLevel !== 'author');
+		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', this.secondsToFormattedString(Math.ceil(existing.remainingTime / 1000)), command.cooldownLevel !== 'author');
+	}
+
+	secondsToFormattedString(time) {
+		const days = Math.floor(time / 86400);
+		const hours = Math.floor(time / 3600);
+		const minutes = Math.floor(time / 3600 / 60);
+		const seconds = Math.floor(time / 3600 % 60);
+		return `${days ? `${days}d ` : ''}${hours ? `${hours}h ` : ''}${minutes ? `${minutes}m ` : ''}${seconds ? `${seconds}s ` : ''}`;
```
Comment (bdistin) [inline src/inhibitors/cooldown.js]: Things should not just be able to pop out of existence if larger things are displayed... Also, this is wonky, try 2 days (172800) you get 2d 48h 48s (no minutes, and 48 seconds out of nowhere)  https://repl.it/repls/GrowingCourageousMisrac And you can't even get accurate seconds results.

**Comment by Skillz4Killz** [comment]
Made the necessary changes to make it work for more than one day interval.
image

File: src/inhibitors/cooldown.js [hunk 4]
```diff
@@ -11,7 +11,15 @@ module.exports = class extends Inhibitor {
...
 
-		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', Math.ceil(existing.remainingTime / 1000), command.cooldownLevel !== 'author');
+		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', this.secondsToFormattedString(Math.ceil(existing.remainingTime / 1000)), command.cooldownLevel !== 'author');
+	}
+
+	secondsToFormattedString(time) {
+		const days = Math.floor(time / 86400);
+		const hours = Math.floor((time % 86400) / 3600);
+		const minutes = Math.floor((time % 86400 % 3600) / 60);
+		const seconds = Math.floor(time % 86400 % 3600 % 60);
```
Comment (bdistin) [inline src/inhibitors/cooldown.js]: 60 is a multiple of both 3600 and 86400.

File: src/inhibitors/cooldown.js [hunk 5]
```diff
@@ -11,7 +11,15 @@ module.exports = class extends Inhibitor {
...
 		const existing = command.cooldowns.get(message.levelID);
 
-		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', Math.ceil(existing.remainingTime / 1000), command.cooldownLevel !== 'author');
+		if (existing && existing.limited) throw message.language.get('INHIBITOR_COOLDOWN', this.secondsToFormattedString(Math.ceil(existing.remainingTime / 1000)), command.cooldownLevel !== 'author');
+	}
+
+	secondsToFormattedString(time) {
+		const days = Math.floor(time / 86400);
+		const hours = Math.floor((time % 86400) / 3600);
+		const minutes = Math.floor((time % 86400 % 3600) / 60);
```
Comment (bdistin) [inline src/inhibitors/cooldown.js]: 3600 is a multiple of 86400.

**Comment by Skillz4Killz** [comment]
Requested changes

**Comment by bdistin** [comment]
ok, can you still address the other critique I made in my other review? 172801 should be `2d 0h 0m 1s`, 172800 should be `2d 0h 0m 0s`, 3000 should be `5m 0s`, 3 should be `3s`...

**Comment by Skillz4Killz** [comment]
Damn I have no idea how I forgot that. It should be fixed now to not remove things if a larger amount is shown. https://repl.it/repls/KeenAuthorizedRom

**Pull request merged**